doc/api/esm.md

```diff
@@ -191,6 +191,12 @@ also be used within `"exports"` to define different package entry points per
 environment, including whether the package is referenced via `require` or via
 `import`.
 
+**Warning**: Introducing the `"exports"` field prevents consumers of a package
+from using any entry points that are not defined, including the `package.json`.
+This will likely be a breaking change. To make the introduction of `"exports"`
+non-breaking, either ensure that every previously supported entry point is
+exported or include an export of the root of the package `"./": "./"`.
```

Comment:
I don't think we should be advising this pattern - if users do this it really defeats the point of exports in not providing any encapsulation.

Any user who is slightly unsure of what to do will read this then just always add the base map - and all the nice static analyzability benefits of packages go back out the window.

I would really much rather users simply don't ship exports than ship it badly.

Reply:
Well I guess the only reason to ship exports with a full map is to provide conditional exports for the main entry point? So maybe the example should reflect that?

Reply:
The thing is - the correct thing to do is just take the time to full enumerate every subpath that might be required by users. This way we avoid exposing eg test folders and build scripts.

Reply:
Test folders and build scripts should be ignored via the [package `files` field](https://docs.npmjs.com/files/package.json#files), or failing that, [`.npmignore`](https://docs.npmjs.com/using-npm/developers.html#keeping-files-out-of-your-package).

A more likely dilemma is trying to avoid exposing internal helpers, as per the discussion on [blacklisting specific paths](https://github.com/jkrems/proposal-pkg-exports/issues/46).

Reply:
Preact recently added `"./": "./"` in exports, and very much exposes scripts and test folders - http://unpkg.com/preact/.

Reply:
As a rule, in all my tooling that works with `"exports"` I explicitly ignore `"./": "./"` entries due to the level of cruft that creeps in otherwise.

Reply:
I agree we should be clear that disabling `"exports"`'s encapsulation is discouraged.  However I think the cat is out of the bag on the `"./": "./"` capability, and therefore it is best we communicate what it means and why it gets used.  Suggested wording coming up next.